DEBUG用にtokenリストを出力する機能実装

nafuka11 opened the pull request:
Fix #9

## やったこと
DEBUG用にtokenリストを出力する機能を追加しました。

`make debug` して、 `./minishell_debug` すると確認できます。

## 出力例
```bash
bash-3.2$ export A="echo hello"
bash-3.2$ ./minishell_debug

minishell > echo a 1>hello ; $A | grep b
[0]type: TOKEN     [echo]
[1]type: TOKEN     [a]
[2]type: IO_NUMBER [1]
[3]type: >         [>]
[4]type: TOKEN     [hello]
[5]type: ;         [;]
[6]type: TOKEN     [$A]
[7]type: |         [|]
[8]type: TOKEN     [grep]
[9]type: TOKEN     [b]
[0]type: TOKEN     [echo]
[0]type: TOKEN     [a]
[0]type: TOKEN     [hello]
[0]type: TOKEN     [echo]
[1]type: TOKEN     [hello]
[0]type: TOKEN     [grep]
[0]type: TOKEN     [b]
```

expanderでもlexerを呼び出しているため、あまり見やすい出力ではないですが、とりあえず確認はできそうかなと思います。
もっと良い出力方法あればぜひ提案いただけると幸いです。

nafuka11 committed:
feat(lexer): print_token_list追加

DEBUG用にtokenリストを出力する

nafuka11 committed:
chore(Makefile): SRCS更新

nafuka11 committed:
feat(lexer): DEBUGフラグが立ってる時tokenリストを標準出力に出力するように

ToYeah commented on srcs/lexer/lexer_print.c:
```diff
@@ -0,0 +1,61 @@
+void		print_token_list(t_token *tokens)
+{
+	int	index;
+
```
仕切り線的なのがあると若干見やすいかなと思いました！
```
	if (tokens)
	{
		ft_putendl_fd("-----------------------", STDOUT_FILENO);
	}
```

nafuka11 replied:
仕切り線、いいですね！

全部に仕切り線を入れると、expanderが分かりづらくなるかもと思い、
lexerとexpanderの間に仕切り線を入れるよう、変更を行いました！

出力は以下のようになります。
```bash
bash-3.2$ export A="echo hello"
bash-3.2$ ./minishell_debug 
minishell > $A | grep e; echo world
[0]type: GENERAL   [$A]
[1]type: |         [|]
[2]type: GENERAL   [grep]
[3]type: GENERAL   [e]
[4]type: ;         [;]
[5]type: GENERAL   [echo]
[6]type: GENERAL   [world]
-----------------------------
[0]type: GENERAL   [echo]
[1]type: GENERAL   [hello]
[0]type: GENERAL   [grep]
[0]type: GENERAL   [e]
hello
[0]type: GENERAL   [echo]
[0]type: GENERAL   [world]
world
```

参考例として、全てに仕切り線を入れると以下のようになります。
```bash
minishell > $A | grep e; echo world
[0]type: GENERAL   [$A]
[1]type: |         [|]
[2]type: GENERAL   [grep]
[3]type: GENERAL   [e]
[4]type: ;         [;]
[5]type: GENERAL   [echo]
[6]type: GENERAL   [world]
-----------------------------
[0]type: GENERAL   [echo]
[1]type: GENERAL   [hello]
-----------------------------
[0]type: GENERAL   [grep]
-----------------------------
[0]type: GENERAL   [e]
-----------------------------
hello
[0]type: GENERAL   [echo]
-----------------------------
[0]type: GENERAL   [world]
-----------------------------
world
```

nafuka11 replied:
lexerとexpanderの仕切りを `=` , expanderの仕切りを `-` にすると見やすそうなので、こちらで行こうと思います。
c055622 で修正を行いました。

```bash
minishell > $A | grep e; echo world
[0]type: GENERAL   [$A]
[1]type: |         [|]
[2]type: GENERAL   [grep]
[3]type: GENERAL   [e]
[4]type: ;         [;]
[5]type: GENERAL   [echo]
[6]type: GENERAL   [world]
================================
[0]type: GENERAL   [echo]
[1]type: GENERAL   [hello]
--------------------------------
[0]type: GENERAL   [grep]
--------------------------------
[0]type: GENERAL   [e]
--------------------------------
hello
[0]type: GENERAL   [echo]
--------------------------------
[0]type: GENERAL   [world]
--------------------------------
world
```

ToYeah replied:
凄くいいと思います！！！！

ToYeah reviewed (approved):
放置してたissue対応ありがとうございます...!
動作問題ないと思います。
今でも十分見やすいと思います！

nafuka11 committed:
style(lexer): print_token_listでlexerとexpanderの間に区切り線を出力するよう変更

esc_flagがFALSE(=lexer)の時区切り線を出力。
lexerは1回しか呼ばれないため、区切り線はlexerとexpanderの間に1つとなる。

nafuka11 committed:
style(lexer): print_token_listでesc_flagに応じて区切り線を変更するように

=: lexerとexpanderの区切り
-: expanderの区切り

nafuka11 committed:
Merge branch 'master' into 9-print-token-for-debug

nafuka11 commented:
コンフリクトの修正を行いました！
